### DISCUSSION TITLE
Autopilot fails to install in a python 3.11 environment, but in a confusing way

### DISCUSSION THREAD

**Discussion by Rodgers-PAC-Lab** [wehr-lab/autopilot#208] [General]
I just set up a new pi with autopilot and Raspbian Lite OS, which ships with python 3.11 now and causes some errors with unhelpful diagnostics so figured I'd let people know.

When you get to the pip install -e .[pilot] step, instead of telling you about the python version conflict it does one of two things: 
If you haven't increased the size of the swapfile it'll just hang forever. 
If you have increased the swapfile it'll take an hour+, start compiling C and fortran (?!) objects related to sci-py, and finally tell you 
```
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

.... So if you're getting either of those symptoms, check your python version. I used pyenv to set up python 3.9, and then it all worked fine.

**Comment by sneakers-the-rat** [comment]
This is likely a longstanding issue with poetry where it didnt handle multiple architectures well (specifically it wouldnt fetch from pypi for x86-64 and piwheels for arm) so you had to compile from source. I am going to spend time addressing the bugs I introduced (sigh) in the last update today and Monday and will see if this had changed

**Reply by Rodgers-PAC-Lab** [reply]
Oof, good luck and thanks.

Another thing I noticed further on in the installation progress is that the hifiberry setup thinks it works when you set up prefs.json, but when you actually try to run autopilot, ALSA can't find the hifiberry. After some troubleshooting I think it's because the setup is adding some dtoverlay things to boot/config.txt. On this new raspbian install, boot/config.txt starts with this warning:

"DO NOT EDIT THIS FILE
The file you are looking for has moved to /boot/firmware/config.txt"

On Monday I'll see if adding the same lines to boot/firmware/config.txt will make the hifiberry work. Not sure when that particular change was implemented in raspbian.

**Reply by sneakers-the-rat** [reply]
oh awesome, good to know they have moved that. will also update.
